astropy/io/fits/file.py

```diff
@@ -400,6 +404,33 @@ def _overwrite_existing(self, overwrite, fileobj, closed):
             else:
                 raise IOError("File {!r} already exists.".format(self.name))
 
+    def _test_for_compression(self, obj_or_name, magic, mode, ext=''):
+        """Attempt to determine if the given file is compressed"""
+        is_compressed = False
+        if ext == '.gz' or magic.startswith(GZIP_MAGIC):
+            # Handle gzip files
+            kwargs = dict(mode=IO_FITS_MODES[mode])
+            if isinstance(obj_or_name, string_types):
+                kwargs['filename'] = obj_or_name
+            else:
+                kwargs['fileobj'] = obj_or_name
+            self._file = gzip.GzipFile(**kwargs)
+            self.compression = 'gzip'
+        elif ext == '.zip' or magic.startswith(PKZIP_MAGIC):
+            # Handle zip files
+            self._open_zipfile(self.name, mode)
+            self.compression = 'zip'
+        elif ext == '.bz2' or magic.startswith(BZIP2_MAGIC):
+            # Handle bzip2 files
+            if mode in ['update', 'append']:
+                raise IOError("update and append modes are not supported "
+                              "with bzip2 files")
+            # bzip2 only supports 'w' and 'r' modes
+            bzip2_mode = 'w' if mode == 'ostream' else 'r'
+            self._file = bz2.BZ2File(obj_or_name, mode=bzip2_mode)
+            self.compression = 'bzip2'
+        return self.compression # Will be 'None' if no compression detected
```

Comment:
Return `False` to be sure, or maybe raise an exception (what was the previous behavior ?)
`is_compressed` is not used.